libs/camutils/include/camutils/Manipulator.h

```diff
@@ -186,14 +206,53 @@ class Manipulator {
     virtual void grabEnd() = 0;
 
     /**
-     * Dollys the camera along the viewing direction.
+     * Keys used to translate the camera in FREE_FLIGHT mode.
+     * UP and DOWN dolly the camera forwards and backwards.
+     * LEFT and RIGHT strafe the camera left and right.
+     */
+    enum class Key {
+        UP,
+        LEFT,
+        DOWN,
+        RIGHT,
+
+        COUNT
+    };
+
+    /**
+     * Signals that a key is now in the down state.
+     *
+     * In FREE_FLIGHT mode, the camera is translated forward and backward and strafed left and right
+     * depending on the depressed keys. This allows WASD-style movement.
+     */
+    virtual void keyDown(Key key);
+
+    /**
+     * Signals that a key is now in the up state.
+     *
+     * @see keyDown
+     */
+    virtual void keyUp(Key key);
+
+    /**
+     * In MAP and ORBIT modes, dollys the camera along the viewing direction.
+     * In FREE_FLIGHT mode, adjusts the move speed of the camera.
      *
      * @param x X-coordinate for point of interest in viewport space
      * @param y Y-coordinate for point of interest in viewport space
      * @param scrolldelta Negative means "zoom in", positive means "zoom out"
      */
     virtual void zoom(int x, int y, FLOAT scrolldelta) = 0;
```

Comment:
Nice, this feature surprised me but then I saw that sketchfab has it too!  Maybe we should rename this method to "scroll" everywhere, since it's not really zooming in free flight mode.  Also add a note that `x` and `y` are ignored in free flight mode.

Reply:
Yup, it seems to be a common pattern. I got the idea from Unreal.